x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/ListConnectorSyncJobsAction.java

```diff
@@ -91,30 +107,39 @@ public boolean equals(Object o) {
             Request request = (Request) o;
             return Objects.equals(pageParams, request.pageParams)
                 && Objects.equals(connectorId, request.connectorId)
-                && connectorSyncStatus == request.connectorSyncStatus;
+                && connectorSyncStatus == request.connectorSyncStatus
+                && connectorSyncJobTypeList == null
+                    ? request.connectorSyncJobTypeList == null
+                    : connectorSyncJobTypeList.equals(request.connectorSyncJobTypeList);
         }
 
         @Override
         public int hashCode() {
-            return Objects.hash(pageParams, connectorId, connectorSyncStatus);
+            return Objects.hash(pageParams, connectorId, connectorSyncStatus, connectorSyncJobTypeList);
         }
 
+        @SuppressWarnings("unchecked")
         private static final ConstructingObjectParser<ListConnectorSyncJobsAction.Request, String> PARSER = new ConstructingObjectParser<>(
             "list_connector_sync_jobs_request",
             p -> new ListConnectorSyncJobsAction.Request(
                 (PageParams) p[0],
                 (String) p[1],
-                p[2] != null ? ConnectorSyncStatus.fromString((String) p[2]) : null
+                p[2] != null ? ConnectorSyncStatus.fromString((String) p[2]) : null,
+                p[3] != null ? ((List<String>) p[3]).stream().map(ConnectorSyncJobType::fromString).toList() : null
             )
         );
 
         static {
             PARSER.declareObject(constructorArg(), (p, c) -> PageParams.fromXContent(p), PAGE_PARAMS_FIELD);
             PARSER.declareString(optionalConstructorArg(), CONNECTOR_ID_FIELD);
             PARSER.declareString(optionalConstructorArg(), ConnectorSyncJob.STATUS_FIELD);
+            PARSER.declareStringArray(optionalConstructorArg(), ConnectorSyncJob.JOB_TYPE_FIELD);
         }
 
         public static ListConnectorSyncJobsAction.Request parse(XContentParser parser) {
+            Logger logger = LogManager.getLogger(Request.class);
+            logger.info("parser: " + parser);
+            logger.info("PARSER: " + PARSER);
```

Comment:
Should this be removed?

Reply:
done: https://github.com/elastic/elasticsearch/pull/104855/commits/453591218b6a008e29ba6302ca52ca2dcd5b742d